.github/scripts/internationalization.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -x
+
+evalrc() {
+  if [ "$1" -ne "0" ]; then
+    echo "$2"
+    exit $1
+  fi
+}
+
+# This function converts candlepin version to a int value with (Zero) padding
+# which helps in comparing versions.
+# Examples -
+# 2.5.6  -> 2005006000
+# 3.2.0  -> 3002000000
+# 2.9.6  -> 2009006000
+get_simplified_version() {
+  echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'
+}
+
+git fetch --all
+
+# Declare an associative array
+declare -A candlepin_java
+
+#Clean the project
+./gradlew clean
+evalrc $? "Project cleaning was not successful."
+
+#To generate the auto generated classes and files
+./gradlew war
+evalrc $? "Compilation was not successful for branch $GIT_BRANCH."
+
+#Execute the gettext task
+./gradlew gettext
+evalrc $? "Gettext was not successful for branch $GIT_BRANCH."
+
+files=$(git diff | egrep -v -e '^( |\+#|\-#|@@|\+\+\+|\-\-\-|diff|index)' -e 'X-Generator' -e 'POT-Creation-Date')
+
+if [ ! -z "$files" ]; then
```

Comment:
IMHO a simpler way could be:
- copy the `keys.pot` file before the message extraction
- after the extraction, compare the old with the new one excluding the creation date

I implemented something like that for virt-manager few years ago:
https://github.com/virt-manager/virt-manager/blob/7cd6151a212c4c477a947fe5a7f2b3363dd0dcbd/.github/workflows/translations.yml#L41-L58